### DISCUSSION TITLE
What new features should Fuse.js have?

### DISCUSSION THREAD

**Discussion by krisk** [krisk/Fuse#526] [Ideas]
Hey all! Fuse.js has been out for many years, and - dare I say - seems to be the most widely used JS fuzzy search library 😄. For the next iteration of the library, I'd love to know your ideas. What are some things you'd love Fuse.js to handle? Thanks! We can upvote/comment on anything that stands out.

**Comment by stephent** [comment]
Not sure if this is supported already - don’t think so - but I have a use case where the ability to do incremental indexing would useful. The source files are not all available at a single point in time. Alternatively, the option to merge indexes would work (and actually could be helpful in addition).

The use case is a next.js site using static site generation from a mix of local markdown files and other blog posts pulled from a CMS. I’d like to generate a single persisted JSON index file at build time, then run search against it either server side or even client side (TBD).

Great library - thank you!

**Reply by krisk** [reply]
Right now you could so by using [`fuse.add`](https://fusejs.io/api/methods.html#add), though I admit the need for looping 😄

**Comment by ghost** [comment]
Thanks for the great library.
If possible, I would like to adjust the keys options etc. when I execute a search query.

**Comment by vwkd** [comment]
Filtering nested object results.

A simple example is only including the properties which are searched. Another example is removing any subarray elements that do not contain any matches, see also https://github.com/krisk/Fuse/issues/690.

**Comment by benjick** [comment]
Hello!

Two features I would want for this library:

1. The possibility to do the equivalent of `fuse.search(searchTerm)` inside an `$and`. Maybe something like this:
```ts
fuse.search({
  $and: [
    {
      $val: searchTerm,
    },
  ],
})
```

2. Searching in ranges. I would want to be able to get results where `year` is between 2020 and 2022 for example

**Reply by vwkd** [reply]
Regarding 2., why don’t you just filter the results?

Adding this to the engine itself would only help performance, but probably not worth the effort for Fuse’s constrained use cases.

**Comment by ishanuda** [comment]
Hi,
1. Would like to see the [Proximity Search](https://en.wikipedia.org/wiki/Proximity_search_(text)) in the library as a built in
2. API for extensibility with the plugins

regards
